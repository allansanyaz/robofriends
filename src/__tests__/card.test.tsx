import React from 'react';
import { screen, render } from '@testing-library/react';
import { describe, expect, test } from "@jest/globals";
import '@testing-library/jest-dom';
import Card from "../components/card";
import { cardData } from "../utils/dummy.data";

describe('Test render of Card component', () => {
    test('Checking to see if the card component is loading', async() => {
        const view = render(<Card {...cardData} />)
        expect.assertions(2);
        // assert that the card component is rendered
        const cardItem = await screen.findByText(cardData.email);
        expect(cardItem.textContent).toEqual(cardData.email);
        expect(view).toMatchSnapshot();
        
    });
});

