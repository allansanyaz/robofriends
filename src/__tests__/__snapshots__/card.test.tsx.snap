// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render of Card component Checking to see if the card component is loading 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
      >
        <img
          alt="robots"
          src="https://robohash.org/1?200x200"
        />
        <div>
          <h2>
            John Doe
          </h2>
          <p>
            john@doe.com
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
    >
      <img
        alt="robots"
        src="https://robohash.org/1?200x200"
      />
      <div>
        <h2>
          John Doe
        </h2>
        <p>
          john@doe.com
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
